go.mod

```diff
@@ -1,43 +1,43 @@
 module github.com/google/osv-scanner
 
-go 1.22.7
+go 1.23
```

Comment:
@another-rex is there any concern to upgrade to Go 1.23?
since deps.dev recently [upgraded to Go 1.23](https://github.com/google/deps.dev/pull/135), so we need to update accordingly to depend on the latest version.

Reply:
Hmm... I think the policy is still to stay on the oldest supported version which is still 1.22, but I think that'll soon be 1.23 since the last release was around 5 months ago, so probably fine to update this.

Can you add the patch version here as well?

Reply:
done